src/build.rs

```diff
@@ -80,20 +81,77 @@ fn is_wasm32_target_in_sysroot(sysroot: &PathBuf) -> bool {
     }
 }
 
-fn check_wasm32_target() -> Result<bool, Error> {
+struct Wasm32Check {
```

Comment:
this logic is getting pretty large- we may consider making build a directory, having the primary build work in a mod.rs and putting the wasm32 target logic in it's own file called wasm-target or something similar

Reply:
Sure, I can move it